src/controller/java/templates/partials/decode_value.zapt

```diff
@@ -0,0 +1,30 @@
+jobject {{asSymbol label}};
+{{#if isOptional}}
+if (!dataResponse.{{asLowerCamelCase label}}.HasValue()) {
+  chip::JniReferences::GetInstance().CreateOptional(nullptr, {{asSymbol label}});
+} else {
+{{#if isNullable}}
+if (dataResponse.{{asLowerCamelCase label}}.Value().IsNull()) {
+  chip::JniReferences::GetInstance().CreateOptional(nullptr, {{asSymbol label}});
+} else {
+{{/if}}
+{{/if}}
+{{~#*inline "item"}}dataResponse.{{asLowerCamelCase name}}{{#if isOptional}}.Value(){{/if}}{{#if isNullable}}.Value(){{/if}}{{/inline}}
+{{#if isArray}}
+{{asSymbol label}} = nullptr; /* Array - Conversion from this type to Java is not properly implemented yet */
+{{else if (isOctetString type)}}
+{{asSymbol label}} = chip::ByteArray(env, {{>item}}).jniValue();
+{{else if (isCharString type)}}
+{{asSymbol label}} = chip::UtfString(env, {{>item}}).jniValue();
+{{else}}
+std::string {{asSymbol label}}ClassName = "java/lang/{{asJavaBasicTypeForZclType type true}}";
```

Comment:
Is there a difference between `{{#if_is_struct type}}` and `{{#if isStruct}}`? I've noticed that ZAP can't seem to find inline partials within the `{{else}}` block of `if_is_struct`, so using `isStruct` for now...

Reply:
Yes, there is a difference.  One is testing something about the type and one is testing something about the context object.  They are not the same test, especially as you dig into types inside of other types (e.g. structs inside lists, structs inside structs, etc), because the `isStruct` thing only gets set on things that `enhanced*` examines....

I don't know what issue you ran into with partials inside `{{else}}`, though I would not be surprised by any sort of random breakage...